plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -3116,6 +3117,86 @@ private Optional<SystemTable> getRawSystemTable(ConnectorSession session, Schema
         };
     }
 
+    @Override
+    public Optional<ConnectorTableHandle> applyDelete(ConnectorSession session, ConnectorTableHandle handle)
+    {
+        DeltaLakeTableHandle tableHandle = (DeltaLakeTableHandle) handle;
+        if (tableHandle.getMetadataEntry().isChangeDataFeedEnabled().orElse(false)) {
+            // For tables with CDF enabled the DELETE operation can't be performed only on metadata files
+            return Optional.empty();
+        }
+        if (!tableHandle.getNonPartitionConstraint().isAll()) {
```

Comment:
This check is probably not necessary - `applyDelete` is not being called when the table handle has a non partition constraint.

Reply:
if this is not necessary then remove it :P